tests/test_adjoint_jacobian.py

```diff
@@ -215,7 +215,7 @@ def test_proj_unsupported(self, dev):
 
         with qml.tape.QuantumTape() as tape:
             qml.CRX(0.1, wires=[0, 1])
-            qml.expval(qml.Projector([0], wires=[0]) @ qml.PauliZ(0))
+            qml.expval(qml.Projector([0], wires=[0]) @ qml.PauliZ(1))
```

Comment:
Got [this failure](https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/8363827153/job/22897751458?pr=649) on this test, as well as the same test from `test_vjp.py`. Why was this test working to begin with?

Reply:
Was happening because the adjoint validation wasn't catching `Projector`s inside `Prod`s, so the C++ layer raised the error in the failure. With `Tensor` the validation error would get raised way before we reached the C++ layer.

Reply:
Can we add a case with overlapping wires if this is supposed to be supported? I guess it should as this simply means multiplying the matrices under the hood.

Reply:
In that case I will have to update the serialization to handle `Prod` with overlapping wires.